### ISSUE TITLE
branch-3.2-8db6910 fe Cannot execute any SQL on external table and reports: ERROR 1064 (HY000): org/apache/arrow/util/Preconditions

### ISSUE THREAD

**Issue by trikker** [StarRocks/starrocks#38830]
### Steps to reproduce the behavior (Required)

after building git branch branch-3.2 and replacing fe dir, cannot execute some SQLs and report: ERROR 1064 (HY000): org/apache/arrow/util/Preconditions

```
'root'@(none) Wed Jan 10 19:10:31 2024>set catalog hive;
Query OK, 0 rows affected (0.00 sec)

'root'@(none) Wed Jan 10 19:10:36 2024>use hivessbtest;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
'root'@hivessbtest Wed Jan 10 19:10:38 2024>select    p_brand,    sum(LO_REVENUE) as revenue_sum  from    P_LINEORDER l    left join part p on l.LO_PARTKEY = p.P_PARTKEY  group by    p_brand limit 1;
ERROR 1064 (HY000): org/apache/arrow/util/Preconditions
'root'@hivessbtest Wed Jan 10 19:10:43 2024>explain select    p_brand,    sum(LO_REVENUE) as revenue_sum  from    P_LINEORDER l    left join part p on l.LO_PARTKEY = p.P_PARTKEY  group by    p_brand;
ERROR 1064 (HY000): org/apache/arrow/util/Preconditions

# simple SQL can succeed
'root'@hivessbtest Wed Jan 10 18:09:39 2024>select * from p_lineorder limit 1;
+-------------+---------------+------------+------------+------------+------------------+-----------------+-------------+------------------+------------------+-------------+------------+---------------+--------+---------------+-------------+-----------+--------------+
| lo_orderkey | lo_linenumber | lo_custkey | lo_partkey | lo_suppkey | lo_orderpriotity | lo_shippriotity | lo_quantity | lo_extendedprice | lo_ordtotalprice | lo_discount | lo_revenue | lo_supplycost | lo_tax | lo_commitdate | lo_shipmode | v_revenue | lo_orderdate |
+-------------+---------------+------------+------------+------------+------------------+-----------------+-------------+------------------+------------------+-------------+------------+---------------+--------+---------------+-------------+-----------+--------------+
|     5984453 |             6 |      26017 |      57718 |        808 | 4-NOT SPECIFIED  |               0 |          44 |          7373124 |         23572676 |           3 |    7151930 |        100542 |      0 |      19931007 | RAIL        |  22119372 |     19930711 |
+-------------+---------------+------------+------------+------------+------------------+-----------------+-------------+------------------+------------------+-------------+------------+---------------+--------+---------------+-------------+-----------+--------------+


# after setting enable_materialized_view_rewrite = false, the SQL can execute

'root'@hivessbtest Wed Jan 10 19:10:47 2024>set enable_materialized_view_rewrite=false;
Query OK, 0 rows affected (0.00 sec)

'root'@hivessbtest Wed Jan 10 19:10:54 2024>select    p_brand,    sum(LO_REVENUE) as revenue_sum  from    P_LINEORDER l    left join part p on l.LO_PARTKEY = p.P_PARTKEY  group by    p_brand limit 1;
+-----------+-------------+
| p_brand   | revenue_sum |
+-----------+-------------+
| MFGR#1519 | 18607392567 |
+-----------+-------------+
1 row in set (0.48 sec)

'root'@hivessbtest Wed Jan 10 19:10:57 2024>explain select    p_brand,    sum(LO_REVENUE) as revenue_sum  from    P_LINEORDER l    left join part p on l.LO_PARTKEY = p.P_PARTKEY  group by    p_brand;
+---------------------------------------------------------+
| Explain String                                          |
+---------------------------------------------------------+
| PLAN FRAGMENT 0                                         |
|  OUTPUT EXPRS:23: p_brand | 28: sum                     |
|   PARTITION: UNPARTITIONED                              |
|                                                         |
|   RESULT SINK                                           |
|                                                         |
|   8:EXCHANGE                                            |
|                                                         |
| PLAN FRAGMENT 1                                         |
|  OUTPUT EXPRS:                                          |
|   PARTITION: HASH_PARTITIONED: 23: p_brand              |
|                                                         |
|   STREAM DATA SINK                                      |
|     EXCHANGE ID: 08                                     |
|     UNPARTITIONED                                       |
|                                                         |
|   7:AGGREGATE (merge finalize)                          |
|   |  output: sum(28: sum)                               |
|   |  group by: 23: p_brand                              |
|   |                                                     |
|   6:EXCHANGE                                            |
|                                                         |
| PLAN FRAGMENT 2                                         |
|  OUTPUT EXPRS:                                          |
|   PARTITION: RANDOM                                     |
|                                                         |
|   STREAM DATA SINK                                      |
|     EXCHANGE ID: 06                                     |
|     HASH_PARTITIONED: 23: p_brand                       |
|                                                         |
|   5:AGGREGATE (update serialize)                        |
|   |  STREAMING                                          |
|   |  output: sum(12: lo_revenue)                        |
|   |  group by: 23: p_brand                              |
|   |                                                     |
|   4:Project                                             |
|   |  <slot 12> : 12: lo_revenue                         |
|   |  <slot 23> : 23: p_brand                            |
|   |                                                     |
|   3:HASH JOIN                                           |
|   |  join op: LEFT OUTER JOIN (BROADCAST)               |
|   |  colocate: false, reason:                           |
|   |  equal join conjunct: 4: lo_partkey = 19: p_partkey |
|   |                                                     |
|   |----2:EXCHANGE                                       |
|   |                                                     |
|   0:HdfsScanNode                                        |
|      TABLE: p_lineorder                                 |
|      partitions=2406/2406                               |
|      cardinality=6001175                                |
|      avgRowSize=2.0                                     |
|                                                         |
| PLAN FRAGMENT 3                                         |
|  OUTPUT EXPRS:                                          |
|   PARTITION: RANDOM                                     |
|                                                         |
|   STREAM DATA SINK                                      |
|     EXCHANGE ID: 02                                     |
|     UNPARTITIONED                                       |
|                                                         |
|   1:HdfsScanNode                                        |
|      TABLE: part                                        |
|      partitions=1/1                                     |
|      cardinality=1                                      |
|      avgRowSize=2.0                                     |
+---------------------------------------------------------+
65 rows in set (0.03 sec)


```

fe.warn.log，I don't why it shows "[CachingMvPlanContextBuilder.loadMvPlanContext():71] load mv plan cache failed: mv2
com.starrocks.sql.analyzer.SemanticException: Getting analyzing error. Detail message: Unknown database 'hiveviewtestdb'.", it always come along with "java.lang.NoClassDefFoundError: org/apache/arrow/util/Preconditions" 

```
2024-01-10 19:06:09,462 WARN (starrocks-mysql-nio-pool-8|1633) [CachingMvPlanContextBuilder.loadMvPlanContext():71] load mv plan cache failed: mv2
com.starrocks.sql.analyzer.SemanticException: Getting analyzing error. Detail message: Unknown database 'hiveviewtestdb'.
        at com.starrocks.sql.analyzer.QueryAnalyzer.resolveTable(QueryAnalyzer.java:1005) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.access$200(QueryAnalyzer.java:97) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.resolveTableRef(QueryAnalyzer.java:296) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:194) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.SelectRelation.accept(SelectRelation.java:242) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:119) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryRelation(QueryAnalyzer.java:134) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:124) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:56) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:119) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.QueryAnalyzer.analyze(QueryAnalyzer.java:107) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.AnalyzerVisitor.visitQueryStatement(AnalyzerVisitor.java:358) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.AnalyzerVisitor.visitQueryStatement(AnalyzerVisitor.java:141) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:56) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:66) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.AnalyzerVisitor.analyze(AnalyzerVisitor.java:143) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:34) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.rule.transformation.materialization.MvUtils.getRuleOptimizedLogicalPlan(MvUtils.java:407) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.MaterializedViewOptimizer.optimize(MaterializedViewOptimizer.java:50) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.MaterializedViewOptimizer.optimize(MaterializedViewOptimizer.java:31) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.MvPlanContextBuilder.getPlanContext(MvPlanContextBuilder.java:36) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.CachingMvPlanContextBuilder.loadMvPlanContext(CachingMvPlanContextBuilder.java:69) ~[starrocks-fe.jar:?]
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2406) ~[caffeine-2.9.3.jar:?]
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853) ~[?:1.8.0_152]
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2404) ~[caffeine-2.9.3.jar:?]
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2387) ~[caffeine-2.9.3.jar:?]
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:108) ~[caffeine-2.9.3.jar:?]
        at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:62) ~[caffeine-2.9.3.jar:?]
        at com.starrocks.sql.optimizer.CachingMvPlanContextBuilder.getPlanContext(CachingMvPlanContextBuilder.java:61) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.MvRewritePreprocessor.filterValidMVs(MvRewritePreprocessor.java:298) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.MvRewritePreprocessor.prepare(MvRewritePreprocessor.java:114) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.prepare(Optimizer.java:273) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:135) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:178) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.planQuery(StatementPlanner.java:151) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:125) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:84) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:486) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:379) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:570) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:848) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_152]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_152]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_152]
2024-01-10 19:06:09,646 WARN (starrocks-mysql-nio-pool-8|1633) [StmtExecutor.execute():669] execute Exception, sql select    p_brand,    sum(LO_REVENUE) as revenue_sum  from    P_LINEORDER l    left join part p on l.LO_PARTKEY = p.P_PARTKEY  group by    p_brand
java.lang.NoClassDefFoundError: org/apache/arrow/util/Preconditions
        at com.starrocks.sql.optimizer.rule.transformation.materialization.rule.BaseMaterializedViewRewriteRule.getQuerySplitPredicate(BaseMaterializedViewRewriteRule.java:208) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.rule.transformation.materialization.rule.BaseMaterializedViewRewriteRule.transform(BaseMaterializedViewRewriteRule.java:132) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.ApplyRuleTask.execute(ApplyRuleTask.java:88) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.task.SeriallyTaskScheduler.executeTasks(SeriallyTaskScheduler.java:65) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.memoOptimize(Optimizer.java:686) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimizeByCost(Optimizer.java:216) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.optimizer.Optimizer.optimize(Optimizer.java:139) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.createQueryPlan(StatementPlanner.java:178) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.planQuery(StatementPlanner.java:151) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:125) ~[starrocks-fe.jar:?]
        at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:84) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:486) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:379) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:570) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:848) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:69) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_152]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_152]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_152]

```

### Expected behavior (Required)

No error.

### Real behavior (Required)

ERROR 1064 (HY000): org/apache/arrow/util/Preconditions

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

```
fe version: branch-3.2-8db6910

'root'@hivessbtest Wed Jan 10 18:01:15 2024>show frontends;
+----------------------------------+--------------+-------------+----------+-----------+---------+----------+------------+------+-------+-------------------+---------------------+----------+--------+---------------------+--------------------+
| Name                             | IP           | EditLogPort | HttpPort | QueryPort | RpcPort | Role     | ClusterId  | Join | Alive | ReplayedJournalId | LastHeartbeat       | IsHelper | ErrMsg | StartTime           | Version            |
+----------------------------------+--------------+-------------+----------+-----------+---------+----------+------------+------+-------+-------------------+---------------------+----------+--------+---------------------+--------------------+
| 10.45.16.207_19010_1704286476648 | 10.45.16.207 | 19010       | 18030    | 19030     | 19020   | LEADER   | 1592310756 | true | true  | 844514            | 2024-01-10 18:02:19 | true     |        | 2024-01-10 17:51:46 | branch-3.2-8db6910 |
| 10.45.16.208_19010_1704286633382 | 10.45.16.208 | 19010       | 18030    | 19030     | 19020   | FOLLOWER | 1592310756 | true | true  | 844513            | 2024-01-10 18:02:19 | true     |        | 2024-01-10 17:57:46 | branch-3.2-8db6910 |
| 10.45.16.206_19010_1704858293669 | 10.45.16.206 | 19010       | 18030    | 19030     | 19020   | FOLLOWER | 1592310756 | true | true  | 844513            | 2024-01-10 18:02:19 | true     |        | 2024-01-10 17:47:18 | branch-3.2-8db6910 |
+----------------------------------+--------------+-------------+----------+-----------+---------+----------+------------+------+-------+-------------------+---------------------+----------+--------+---------------------+--------------------+
3 rows in set (0.04 sec)


'root'@hivessbtest Wed Jan 10 18:01:05 2024>select current_version()
    -> ;
+-------------------+
| current_version() |
+-------------------+
| 3.2.2-269e832     |
+-------------------+
1 row in set (0.01 sec)
```

**Comment by kevincai** [comment]
make a complete replacement of the fe/lib directory, not the copy-and-overwrite-if-exist

**Comment by LiShuMing** [comment]
fixed by : 
https://github.com/StarRocks/starrocks/pull/38845

**Comment by trikker** [comment]
> fixed by : #38845

Thanks.
